capstone/src/main/java/com/google/step/data/Restaurant.java

```diff
@@ -14,15 +14,13 @@
 
 package com.google.step.data;
 
-import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.GeoPt;
-import java.lang.IllegalArgumentException;
 import java.util.ArrayList;
 import java.util.List;
 
 /** A restaurant to be displayed. */
 public final class Restaurant {
-  private final long restaurantKey;
+  private long restaurantKey;
```

Comment:
Ok, I see the issues here. When I looked into it it didn't seem like Optionals are typically used as instance variables, so I instead made the restaurantKey a final Long and had the getRestaurantKey return an Optional<Long>. Then, like you said, new restaurants are passed without a key, and in all other cases they are passed with keys. Please let me know if I got the right idea!

Reply:
Yeah, sorry about that, as I was the one to recommend that solution. I had forgotten that both the `RestaurantClient` and `Restaurant` built `Restaurant` objects, which makes that particular solution quite hairy. I think the reason that my alarm bells were going off is because its seems misleading to call the constructor with `0` as a restaurantKey, as that is not what the restaurantKey is.